### ISSUE TITLE
Code crushes with generated mesh generator mesh and --distributed-mesh with multiple cores

### ISSUE THREAD

**Issue by shoheiogawa** [idaholab/moose#14018]
## Bug Description

While I have been working on a side neighbor marker development, I realize that the code crushes if I use generated mesh generator mesh and `--distributed-mesh` while just having `parallel_type = DISTRIBUTED` or replicated mesh is fine. This doesn't happen if I use a regular mesh block.

## Steps to Reproduce

Code: https://github.com/shoheiogawa/moose/tree/side_neighbor_marker_11643 (5086327)

This adds a side neighbor marker and some input files. The code is under `framework` directory and the input files are at: `test/tests/markers/side_neighbor_marker/sideset_neighbor_marker.i`. After compiling and moving to the input file directory, executing
```
mpiexec -np 2 ../../../moose_test-opt -i sideset_neighbor_marker.i --distributed-mesh
```
causes this error.
```
[0] ./include/libmesh/remote_elem.h, line 119, compiled Sep 10 2019 at 07:36:46
[1] ./include/libmesh/remote_elem.h, line 119, compiled Sep 10 2019 at 07:36:46

application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 1
```

If we have `parallel_type = DISTRIBUTED` without `--distributed-mesh`, the code doesn't cause the error. I.e.,
```
mpiexec -np 2 ../../../moose_test-opt -i sideset_neighbor_marker.i
```

If we have a regular mesh block, the problem doesn't happen.

Having
```
[Mesh]
  # GeneratedMesh works
  type = GeneratedMesh
  dim = 2
  nx = 4
  ny = 4
  parallel_type = DISTRIBUTED
[]
```
works with `--distributed-mesh`.

## Impact

This problem doesn't allow me to use the marker in some cases.

**Comment by lindsayad** [comment]
See #14061 and eff173c in particular. Currently using:
1. `MeshGenerators` 
2. `Adaptivity`
3. `DistributedMesh`
fails because the relationship managers necessary to handle Adaptivity are getting added to a `MeshBase` object that isn't even used in the simulation. This will be fixed once #14061 is merged.

**Comment by shoheiogawa** [comment]
After #14061 is merged, the code doesn't crash anymore.
